pkg/trace/obfuscate/json.go

```diff
@@ -69,7 +69,7 @@ func newJSONObfuscator(cfg *config.JSONObfuscationConfig, o *Obfuscator) *jsonOb
 
 func sqlObfuscationTransformer(o *Obfuscator) func(string) string {
 	return func(s string) string {
-		result, err := o.ObfuscateSQLString(s)
+		result, err := o.ObfuscateSQLString(s, config.SQLObfuscationConfig{})
```

Comment:
```suggestion
		result, err := o.ObfuscateSQLString(s, o.opts.SQLObfuscation)
```

Reply:
I'm sorry but this is very confusing and unacceptable. Please find an alternative implementation which has a single source of truth for the config. Try to aim for something like:
```go
func ObfuscateSQLString(sql string, opts ...) string {
```
And then use that.

Reply:
That's a good point. Since we're sending it as a parameter it looks like we should be able to remove it from the static config, however, then we have no way to specify this in the json obfuscator. We need to think through this a bit more

Reply:
Updated: https://github.com/DataDog/datadog-agent/pull/8556/commits/2354a436ec591c7cbf376a72af4ef4ee95704c6d